.travis-cerrors.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# This is a standalone script so we terminate as soon as anything
+# errors. See https://github.com/travis-ci/travis-ci/issues/1066
+set -e
+set -x
+
+export PATH=$PATH:~/.cargo/bin
+
+RUSTFMT_CONFIG_DIR=$DIR/rust_src
+
+echo "Checking compile errors"
+cd "$DIR/rust_src"
+cargo build --features compile-errors 2> err.out || true
+cat err.out
+grep -q 'compile_error!("error 001");' err.out
```

Comment:
Are you expecting this script to exit with a non-zero exit code in some circumstances? I can't see where that could happen.

Reply:
(`grep` exits with an error code when no hit is found)

Reply:
Would you prefer explicit `|| exit 1` code? I think that's easier to read, but tried not to modify the other script except where necessary.

Reply:
I thought that `grep -q` always returned a zero exit code, but I was wrong. `grep -q` only changes the exit code when there's an error.

That said, could we simplify this to using the exit code of `cargo build` instead?

Reply:
> That said, could we simplify this to using the exit code of cargo build instead?
I don't think so. The idea was to make sure we got an error message originating from the `compile_error!` line, rather than one attributed to the macro (which is quite unhelpful). Thus the grep.